project/Settings.scala

```diff
@@ -274,7 +274,15 @@ object Settings {
   )
 
   def slickScalacSettings = Seq(
-    scalacOptions ++= List("-deprecation", "-feature", "-unchecked", "-Xfuture"),
+    scalacOptions ++= List("-deprecation", "-feature", "-unchecked"),
+    scalacOptions ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, v)) if v <= 12 =>
+          Seq("-Xfuture")
```

Comment:
Wasn't `-Xfuture` deprecated in 2.12 already?
Also do we need this scalac option at all for 2.11? I am curious what feature we are using which needs this flag...

Reply:
Someone could dig into this in a separate PR.